### DISCUSSION TITLE
Tree API Docs

### DISCUSSION THREAD

**Discussion by manunamz** [mar10/wunderbaum#87] [Q&A] [answered]
Hi -- thanks so much for making fancy tree and refactoring it to wunderbaum!

I'm currently trying to setup basic file i/o operations like create new file/folder, move, rename, delete, etc. But I'm struggling a bit with how to implement some of them...Namely how to add/delete/manipulate nodes in the tree and have them render appropriately.

I think I'm basically asking for the ["Tree API" doc page](https://mar10.github.io/wunderbaum/#/tutorial/tutorial_api?id=mutation) to be filled in (currently marked as "under construction").  (Also found [editing grid cells page](https://mar10.github.io/wunderbaum/#/tutorial/tutorial_grid?id=editing) -- is this related?)

Also related:
- Older docs [here](https://mar10.github.io/wunderbaum/api/classes/wunderbaum.Wunderbaum.html).
- Most relevant code in [`/src/wunderbaum.ts`](https://github.com/mar10/wunderbaum/blob/42b089c/src/wunderbaum.ts#L1781) (I think).

**Comment by mar10** [comment, marked as answer]
Hi,
maybe the [tree.applyCommand()](https://mar10.github.io/wunderbaum/api/classes/wunderbaum.Wunderbaum.html#applyCommand) and [node.applyCommand()](https://mar10.github.io/wunderbaum/api/classes/wb_node.WunderbaumNode.html#applyCommand) help. 
From the API docs you can also [jump to the code](https://github.com/mar10/wunderbaum/blob/42b089c/src/wunderbaum.ts#L824) for some details.

**Reply by manunamz** [reply]
Thanks! That helps narrow it down a lot! 🙏
